middleware.go

```diff
@@ -4,22 +4,19 @@ import (
 	"net/http"
 )
 
-// NextMiddlewareFunc is the function signature for the next parameter in ServerHTTPMiddleware
-type NextMiddlewareFunc func(http.ResponseWriter, *http.Request)
```

Comment:
I thought it would just wrap the function signature and make it easier to write internally, but turns out Go's strict typing means functions that have the `func(http.ResponseWriter, *http.Request, func(http.ResponseWriter, *http.Request))` signature can't be used unless the final argument is explicitly wrapped with a `powermux.NextMiddlewareFunc` cast